x-pack/plugins/cases/public/containers/use_get_cases.tsx

```diff
@@ -125,157 +38,40 @@ export const initialData: Cases = {
 };
-export interface UseGetCases extends UseGetCasesState {
-  dispatchUpdateCaseProperty: ({
-    updateKey,
-    updateValue,
-    caseId,
-    version,
-    refetchCasesStatus,
-  }: UpdateCase) => void;
-  refetchCases: () => void;
-  setFilters: (filters: Partial<FilterOptions>) => void;
-  setQueryParams: (queryParams: Partial<QueryParams>) => void;
-  setSelectedCases: (mySelectedCases: Case[]) => void;
-}
 
-const empty = {};
 export const useGetCases = (
   params: {
-    initialQueryParams?: Partial<QueryParams>;
-    initialFilterOptions?: Partial<FilterOptions>;
+    queryParams?: Partial<QueryParams>;
+    filterOptions?: Partial<FilterOptions>;
   } = {}
-): UseGetCases => {
-  const { initialQueryParams = empty, initialFilterOptions = empty } = params;
-  const [state, dispatch] = useReducer(dataFetchReducer, {
-    data: initialData,
-    filterOptions: {
-      ...DEFAULT_FILTER_OPTIONS,
-      ...initialFilterOptions,
-    },
-    isError: false,
-    loading: [],
-    queryParams: { ...DEFAULT_QUERY_PARAMS, ...initialQueryParams },
-    selectedCases: [],
-  });
+): UseQueryResult<Cases> => {
   const toasts = useToasts();
-  const didCancelFetchCases = useRef(false);
-  const didCancelUpdateCases = useRef(false);
-  const abortCtrlFetchCases = useRef(new AbortController());
-  const abortCtrlUpdateCases = useRef(new AbortController());
-
-  const setSelectedCases = useCallback((mySelectedCases: Case[]) => {
-    dispatch({ type: 'UPDATE_TABLE_SELECTIONS', payload: mySelectedCases });
-  }, []);
-
-  const setQueryParams = useCallback((newQueryParams: Partial<QueryParams>) => {
-    dispatch({ type: 'UPDATE_QUERY_PARAMS', payload: newQueryParams });
-  }, []);
-
-  const setFilters = useCallback((newFilters: Partial<FilterOptions>) => {
-    dispatch({ type: 'UPDATE_FILTER_OPTIONS', payload: newFilters });
-  }, []);
-
-  const fetchCases = useCallback(
-    async (filterOptions: FilterOptions, queryParams: QueryParams) => {
-      try {
-        didCancelFetchCases.current = false;
-        abortCtrlFetchCases.current.abort();
-        abortCtrlFetchCases.current = new AbortController();
-        dispatch({ type: 'FETCH_INIT', payload: 'cases' });
-
-        const response = await getCases({
-          filterOptions,
-          queryParams,
-          signal: abortCtrlFetchCases.current.signal,
-        });
-
-        if (!didCancelFetchCases.current) {
-          dispatch({
-            type: 'FETCH_CASES_SUCCESS',
-            payload: response,
-          });
-        }
-      } catch (error) {
-        if (!didCancelFetchCases.current) {
-          if (error.name !== 'AbortError') {
-            toasts.addError(
-              error.body && error.body.message ? new Error(error.body.message) : error,
-              { title: i18n.ERROR_TITLE }
-            );
-          }
-          dispatch({ type: 'FETCH_FAILURE', payload: 'cases' });
-        }
-      }
+  return useQuery(
+    [CASE_LIST_CACHE_KEY, 'cases', params],
```

Comment:
Do we need the second item on the array? `[CASE_LIST_CACHE_KEY, params]` should be enough, right?